.github/workflows/release.yml

```diff
@@ -4,24 +4,29 @@ on:
   release:
     types: [published]
 
+permissions:
+  contents: write
+  packages: write
+
 jobs:
-  publish:
+  goreleaser:
     runs-on: ubuntu-latest
-
     steps:
-    - name: install golang
-      uses: actions/setup-go@v5
-      with:
-        go-version: 1.23.1
-
-    - name: checkout code
-      uses: actions/checkout@v4
-
-    - name: build assets
-      run: make release VERSION=${{ github.event.release.tag_name }}
-
-    - name: gh login
-      run:  echo ${{ secrets.GITHUB_TOKEN }} | gh auth login --with-token
-
-    - name: upload assets to release
-      run: gh release upload ${{ github.event.release.tag_name }} dist/*
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.23.1
```

Comment:
You can link the go.mod instead: https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file